### PR TITLE
Add CMake export infrastructure

### PR DESCRIPTION
**Pull request by matz-e**
Currently will export a target `nmodl::nmodl` for the NMODL binary, and a `nmodl_BINARY` variable for the filename.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +39,7 @@ add_dependencies(nmodl nmodl_copy_python_files nmodl_copy_solver_files)
 cpp_cc_configure_sanitizers(TARGET nmodl)
 
 # =============================================================================
-# Add dependency with nmodl pytnon module (for consumer projects)
+# Add dependency with nmodl python module (for consumer projects)
```
Comment (1uc) [inline src/CMakeLists.txt:42]:
```suggestion
# Add dependency with nmodl Python module (for consumer projects)
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -24,6 +24,7 @@ include_directories(${PYBIND11_INCLUDE_DIR} ${PYTHON_INCLUDE_DIRS})
 # Add executables
 # =============================================================================
 add_executable(nmodl main.cpp)
+set_target_properties(nmodl PROPERTIES PUBLIC_HEADER nmodl.hpp)
```
Comment (1uc) [inline src/CMakeLists.txt:27]: I see it used to get installed and I like one change at a time, so no need for changes. However, did you happen to find out why we need this? To my knowledge NMODL is an application (not library) and a Python module. Hence, it's not obvious to me why we need this.

**Review by 1uc — COMMENTED** [review]
Have you tested integration into NRN?

File: src/CMakeLists.txt (hunk 2 above)
Comment (matz-e) [reply src/CMakeLists.txt:27]: Indeed, it seems to be only used by the unit tests. I can try to remove it, together with adding some "legacy" glue.

Comment (1uc) [reply src/CMakeLists.txt:27]: No legacy glue please.

**Comment by matz-e** [comment]
Tested in the NRN repo, seems to run fine.

**Review by 1uc — APPROVED** [review]

**Pull request merged**